File: frontend/project-gerenciador-tarefas/src/app/services/task.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Task } from '../models/task.model';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class TaskService {
  
  private apiUrl = `${environment.apiUrl}/tasks`;

  constructor(private http: HttpClient) {}

  listar(): Observable<Task[]> {
    return this.http.get<Task[]>(this.apiUrl);
  }

  inserir(task: Task): Observable<Task> {
    return this.http.post<Task>(this.apiUrl, task, { observe: 'body' });
  }

  buscarPorId(id: number): Observable<Task> {
    return this.http.get<Task>(`${this.apiUrl}/${id}`);
  }

  atualizar(task: Task): Observable<Task> {
    return this.http.put<Task>(`${this.apiUrl}/${task.id}`, task, { observe: 'body' });
  }

  excluir(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: backend/gerenciador-tarefas/src/main/java/com/gerenciador_tarefas/service/TaskService.java
package com.gerenciador_tarefas.service;

import java.util.List;

import com.gerenciador_tarefas.model.Task;


public interface TaskService {
	
    public List<Task> listar();

    public Task buscarPorId(Long id);

    public Task salvar(Task task);

    public Task atualizar(Long id, Task taskAtualizado);

    public void excluir(Long id);

}
